app/tree/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -1,44 +1,151 @@
-    componentDidMount () {
-      console.log(this.props)
-      let data = Processors.processCsvData(testdata)
-      console.log(mockPolygonData)
-      const polygonData = Processors.processGeojson(mockPolygonData)
-      console.log(polygonData)
-      data += polygonData
-      const dataset = {
-        data: polygonData,
-        info: {
-          id: 'test'
+export const treeMap: React.FunctionComponent<{}> = () => {
+  const counter:any = useSelector(state => state);
+  const dispatch = useDispatch();
+  
+  //const blockQuery = useQuery<BlocksQuery, BlocksQuery_getBlockGeometries>(blckQuery, {});
+  const boroughQuery = useQuery<BoroughsQuery, BoroughsQuery_getBoroughGeometries>(brghQuery, {});
+  
+  const [getNtasByBoroughUri, NTAQuery] = useLazyQuery<NtasByBoroughQuery, NtasByBoroughQueryVariables>(ntaQuery);
+  const [getBlocksByNtaUri, BlockQuery] = useLazyQuery<BlocksByNtaQuery, BlocksByNtaQueryVariables>(blckQuery);
+  const [getResult, ResultQuery] = useLazyQuery<ResultsQuery, ResultsQueryVariables>(rsltQuery);
+  
+  function addTreeData(treeData) {
+    const trees = treeData.map(tree => {
+      const point = "POINT (" + tree.longitude + " " + tree.latitude + ")"
+      return {
+        "type": "Feature",
+        "geometry": wkt.parse(point),
+        "properties": {
+          uri: tree.uri,
+          type: "tree",
+          child: null
+        } 
+      }
+    })
+    const featureData = {
+      "type": "FeatureCollection",
+      "features": trees
+    }
+    console.log(featureData)
+    const dataset = {
+      data: Processors.processGeojson(featureData),
+      info: {
+        id: "tree"
+      }
+    }
+    dispatch(addDataToMap({ datasets: dataset, options: {centerMap: true, readOnly: true}}))
+  }
+
+  async function addGeometryData(dataQuery, type: String, child: String){
+    const features = dataQuery.map(feature => {
+      return {
+        "type": "Feature",
+        "geometry": wkt.parse(feature.geometry.wkt),
+        "properties": {
+          "uri": feature.uri,
+          "label": feature.geometry.label,
+          "type": type,
+          "child": child
         }
       }
-      this.props.dispatch(addDataToMap({ datasets: dataset, options: {centerMap: true, readOnly: true}}))
-    }  
+    })
+    const featuredata = {
+      "type": "FeatureCollection",
+      "features": features
+    }
+    console.log(featuredata)
+    const dataset = {
+      data: Processors.processGeojson(featuredata),
+      info: {
+        id: type
+      }
+    }
+    await(dispatch(addDataToMap({ datasets: dataset, options: {centerMap: true, readOnly: true}})))
+  }
 
-    render () {
-     return (
-      <div>
+  const [boroughRendered, setBoroughRender] = useState(false)
+  const [ntaRendered, setNtaRender] = useState(false)
+  const [blockRendered, setBlockRender] = useState(false)
+  const [treeRendered, setTreeRender] = useState(false)
+
+  useEffect(() => {
```

Comment:
I don't think you really need this at all. Use state if you need to keep track of what's rendered.

Reply:
Modifications to redux are not "effects". Effects are things that go outside the React universe, such as modifying the browser DOM directly instead of rendering into the virtual DOM.